### PR TITLE
Added CLI option to read license file for inlets operator.

### PR DESCRIPTION
**Pull request by codeanit**
Signed-off-by: Anit <codeanit@gmail.com>

CLI option `license-file` or `lf` added to read a inlets operator license file.

## Description

CLI option `license-file` or `lf` added to read a inlets operator license file.

## Motivation and Context

- [x] I have raised an issue to propose this change ([64](https://github.com/alexellis/arkade/issues/64))
The existing and previous option to read license key passed as a string is insecure. Hence, this featured was needed.

## How Has This Been Tested?

- Module has been developed as an standalone package
- Unit test has been added to verify it works standalone.
- Manual test are done by running the application is done.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [ ] New application can be installed on ARM platform.
- [ ] New application shows an error message if the architecture is not supported.

### CODE REVIEW

**Comment by codeanit** [comment]
@Waterdrips Hope things are well. Could you please review?

**Comment by alexellis** [comment]
Thanks for the PR. @Waterdrips did you see this ping?

File: pkg/license/license.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Package license reads a license file and
```
Comment (alexellis) [inline pkg/license/license.go]: I don't think we need this package. Can you delete it please?

ioutil.Readfile is fine.

File: pkg/license/license_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package license
```
Comment (alexellis) [inline pkg/license/license_test.go]: Not required, please delete.

File: cmd/apps/inletsoperator_app.go [hunk 3]
```diff
@@ -161,6 +162,15 @@ func MakeInstallInletsOperator() *cobra.Command {
 			overrides["inletsProLicense"] = val
 		}
 
+		if licenseFile, _ := command.Flags().GetString("license-file"); len(licenseFile) > 0 {
```
Comment (alexellis) [inline cmd/apps/inletsoperator_app.go]: Please just use https://golang.org/pkg/io/ioutil/#ReadFile

File: cmd/apps/inletsoperator_app.go [hunk 4]
```diff
@@ -36,10 +37,10 @@ func MakeInstallInletsOperator() *cobra.Command {
 	inletsOperator.Flags().StringP("token-file", "t", "", "Text file containing token or a service account JSON file")
 	inletsOperator.Flags().StringP("secret-key-file", "s", "", "Text file containing secret key, used for providers like ec2")
 	inletsOperator.Flags().Bool("update-repo", true, "Update the helm repo")
-
 	inletsOperator.Flags().String("pro-client-image", "", "Docker image for inlets-pro's client")
 	inletsOperator.Flags().Bool("helm3", true, "Use helm3, if set to false uses helm2")
 	inletsOperator.Flags().StringArray("set", []string{}, "Use custom flags or override existing flags \n(example --set=image=org/repo:tag)")
+	inletsOperator.Flags().String("license-file", "lf", "The text file containing license key for inlets-pro")
```
Comment (alexellis) [inline cmd/apps/inletsoperator_app.go]: Not sure you can use a two letter alias of `lf`, was this tested?

File: pkg/license/license.go (hunk 1 above)
Comment (codeanit) [reply pkg/license/license.go]: Noted! 
@alexellis, I will make the changes as recommended. Thanks.

File: pkg/license/license_test.go (hunk 2 above)
Comment (codeanit) [reply pkg/license/license_test.go]: Noted! 
Thanks @alexellis.

File: cmd/apps/inletsoperator_app.go (hunk 4 above)
Comment (codeanit) [reply cmd/apps/inletsoperator_app.go]: @alexellis, It was tested passing value to the parameter name. I will double check while making changes. Thanks.

File: cmd/apps/inletsoperator_app.go (hunk 3 above)
Comment (codeanit) [reply cmd/apps/inletsoperator_app.go]: @alexellis Thanks. Will do.

File: cmd/apps/inletsoperator_app.go (hunk 4 above)
Comment (alexellis) [reply cmd/apps/inletsoperator_app.go]: Let's set it to something else, a single character. is `-f` available? or `-l`?

Comment (codeanit) [reply cmd/apps/inletsoperator_app.go:43]: Sure. Thanks.

**Comment by alexellis** [comment]
Thanks for having a go at this PR, we needed it a little sooner, but you were not to know.

Fixed via: #83

Please feel free to stop by again and we'll see if we can find something else for you to get a quick win with.

**Comment by codeanit** [comment]
Will do @alexellis. Thanks.

**Pull request closed without merging**